src/Data/FunctorF.purs

```diff
@@ -0,0 +1,17 @@
+module Data.FunctorF where
+
+import Data.NaturalTransformation (type (~>))
+
+-- | A type class for higher order functors; types of kind `(* -> *) -> * -> *`.
+-- |
+-- | The mapping operation `hoist` allows the inner functor to be changed. This
+-- | is similar to `map` for a `Functor`, only here we are transforming a
+-- | functor rather than a value, so correspondingly we perform this with a
+-- | natural transformation rather than a rank-1 function.
+-- |
+-- | Instances must satisfy the following laws:
+-- |
+-- | - Identity: `hoist id = id`
+-- | - Composition: `hoist (f <<< g) = hoist f <<< hoist g`
+class FunctorF t where
+  hoist :: forall f g. (f ~> g) -> t f ~> t g
```

Comment:
That's true, but looking at the search, we don't have any types like those aside from `hoistActionT`, which has the comment about it not being an `MFunctor` for that reason.

`MFunctor`'s hoist only constrains the `m`, not the `n`:
```
hoist :: (MFunctor t, Monad m) => (forall a. m a -> n a) -> t m b -> t n b
```
Which is interesting, because surely then you can hoist a type into something that is un-hoistable?